test/taskcluster_secrets_api_test.js

```diff
@@ -0,0 +1,150 @@
+suite("TaskCluster-Secrets", () => {
+  var helper = require('./helper');
+  var assert = require('assert');
+  var slugid = require('slugid');
+
+  let testValueExpires  = {secret: {data: "bar"}, expires: "2066-10-06T07:25:54.957Z"};
+  let testValueExpires2 = {secret: {data: "foo"}, expires: "2066-10-06T07:25:54.957Z"};
+  let testValueExpired  = {secret: {data: "bar"}, expires: "2011-04-01T00:00:00.000Z"};
+
+  const FOO_KEY = slugid.v4();
+  const BAR_KEY = slugid.v4();
+
+  let testData = [
+    // The "Captain" clients
+    {
+      testName:   "Captain, write allowed key",
+      clientName: "captain-write",
+      apiCall:    "set",
+      name:        "captain:" + FOO_KEY,
+      args:       testValueExpires,
+      res:        {}
+    },
```

Comment:
If you add the test case an extra time... that should ensure that you support re-applying the same put.
```js
   {
     testName:   "Captain, write allowed key",
     clientName: "captain-write",
     apiCall:    "set",
     name:        "captain:" + FOO_KEY,
     args:       testValueExpires,
     res:        {}
   },
```